packages/block-library/src/query-title/edit.js

```diff
@@ -10,45 +10,69 @@ import classnames from 'classnames';
 import {
 	AlignmentControl,
 	BlockControls,
-	useBlockProps,
 	Warning,
+	useBlockProps,
 } from '@wordpress/block-editor';
-import { __ } from '@wordpress/i18n';
+import { __, _x } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import HeadingLevelDropdown from '../heading/heading-level-dropdown';
 
-const SUPPORTED_TYPES = [ 'archive' ];
+const SUPPORTED_TEMPLATES = [ 'archive', 'search', '404' ];
 
 export default function QueryTitleEdit( {
-	attributes: { type, level, textAlign },
+	attributes: { content, level, textAlign },
 	setAttributes,
+	context: { templateSlug },
 } ) {
 	const TagName = `h${ level }`;
 	const blockProps = useBlockProps( {
-		className: classnames( {
+		className: classnames( 'wp-block-query-title__placeholder', {
 			[ `has-text-align-${ textAlign }` ]: textAlign,
-			'wp-block-query-title__placeholder': type === 'archive',
 		} ),
 	} );
-	// The plan is to augment this block with more
-	// block variations like `Search Title`.
-	if ( ! SUPPORTED_TYPES.includes( type ) ) {
+
+	// Infer title content from template slug context
+	// Defaults to content attribute prop
+	switch ( templateSlug ) {
+		case 'archive':
+			// translators: Title for archive template.
+			content = _x( 'Archive title', 'archive template title' );
+			break;
+		case 'search':
+			// translators: Title for search template with dynamic content placeholders.
+			content = _x(
+				// '%total% results found for "%search%"', // Displays number of search results
```

Comment:
This comment doesn't look right

Reply:
Yeah.. it's a line with a comment that's then been completely commented out 😅  I mainly left it in as an alternative search title, but we should probably just pick one to go with for now. To confirm, the two options at the moment are:

- Search results for [search term]
- [total number] results found for [search term]

The second option was taken from the original PR for the search title, so I didn't want to completely dismiss it.

Reply:
I'll remove the commented out option for now so it's less confusing when reading the code.